lib/config/dynamic/merger.go

```diff
@@ -31,6 +31,28 @@ func NewMerger() Merger {
 // Merge merges two config objects together. For now only
 // whitelist registries are merged.
 func (m *merger) Merge(orig, other *config.VirtualContainerHostConfigSpec) (*config.VirtualContainerHostConfigSpec, error) {
+	// merge strategy:
+	//
+	// origWl empty, otherWl empty => empty
+	//
+	// origWl empty, otherWl not empty => otherWl
+	//
+	// origWl not empty, otherWl empty => origWl
+	//
+	// origWl not empty, otherWl not empty => merge result
+	// in thie case, the merge is a set union of origWl
+	// and otherWl, with the following criteria:
+	//
+	// 1. an entry in otherWl cannot make another
+	//    entry in origWl more permissive, e.g.
+	//    foo.docker.io in origWl and *.docker.io
+	//    in otherWl, would not merge
+	// 2. the resulting whitelist should not have
```

Comment:
I'm not sure about this. I think the following would be legitimate:
1. original whitelist: `*.eng.vmware.com`
2. otherWl: `sf.eng.vmware.com`, `aus.eng.vmware.com`

I'm sure that @sflxn added code that will provide a lot of this logic in order to check user supplied registries against the defined whitelist during `pull`. In this case we have a set of user-defined registries that we need to validate but I think we can likely do so iteratively. Don't know about handling for domain wildcards in the otherWl.

I do suggest you make use of the unit tests he added as the easiest way to validate my above assertion.